File: packages/expo-media-library/ios/next/objects/asset/AssetMetadata.swift
import ExpoModulesCore

struct AssetMetadata: Record {
  @Field var id: String
  @Field var creationTime: Int?
  @Field var duration: Int?
  @Field var filename: String?
  @Field var height: Int?
  @Field var width: Int?
  @Field var mediaType: MediaTypeNext = .UNKNOWN
  @Field var modificationTime: Int?
  @Field var isFavorite: Bool

  init () {}

  init(
    id: String,
    creationTime: Int? = nil,
    duration: Int? = nil,
    filename: String?,
    height: Int?,
    width: Int?,
    mediaType: MediaTypeNext,
    modificationTime: Int? = nil,
    isFavorite: Bool
  ) {
    self.id = id
    self.creationTime = creationTime
    self.duration = duration
    self.filename = filename
    self.height = height
    self.width = width
    self.mediaType = mediaType
    self.modificationTime = modificationTime
    self.isFavorite = isFavorite
  }
}

File: packages/expo-media-library/src/types/AssetMetadata.ts
import type { MediaType } from './MediaType';

/**
 * Lightweight metadata for a single asset, returned by [Query.exeForMetadata](#exeformetadata).
 *
 * Contains fields that can be read cheaply from the media store, without resolving file paths or
 * decoding files. Use [Asset](#asset) getters when you need heavier fields such as URI or EXIF data.
 *
 * > On Android, `width` and `height` may be `null` when the media store does not record them.
 */
export type AssetMetadata = {
  id: string;
  filename: string | null;
  mediaType: MediaType;
  width: number | null;
  height: number | null;
  duration: number | null;
  creationTime: number | null;
  modificationTime: number | null;
  isFavorite: boolean;
};
